### PR TITLE
Add min and max methods...

### PR DESCRIPTION
**Pull request by AKushWarrior**
Add minimum and maximum methods for Comparable-based Iterables.

### CODE REVIEW

**Comment by AKushWarrior** [comment]
Given that you already have the typical searching/sorting toolkit, min/max seems like a logical extension...

File: lib/src/list.dart [hunk 1]
```diff
@@ -153,4 +153,66 @@ extension ListX<E> on List<E> {
...
+  ///
+  /// The [compare] function must act as a [Comparator].
+  ///
+  /// The default implementation uses [Comparable.compare] if
+  /// [compare] is omitted.
+  ///
+  /// This Iterable should be composed of Comparables.
+  ///
+  /// If this.length == 0, null will be returned.
+  dynamic min({int Function(dynamic a, dynamic b) compare}) {
```
Comment (passsy) [inline lib/src/list.dart]: should return `E`, not `dynamic`

**Comment by passsy** [comment]
What's wrong with the Iterable extensions [`minBy`](https://github.com/leisim/dartx/blob/da180dd4033c6460294e29a810c0a7ef3a18cc0e/lib/src/iterable.dart#L344), [`maxBy`](https://github.com/leisim/dartx/blob/da180dd4033c6460294e29a810c0a7ef3a18cc0e/lib/src/iterable.dart#L357) and friends?

**Comment by AKushWarrior** [comment]
whoops, you're right. my bad...

**Pull request closed without merging**